### PR TITLE
Docker support

### PR DESCRIPTION
**Pull request by julien-noblet**
Bonjour!
Comme promis au SOTMFR, voici un Dockerfile simplifié au max mais efficace.

En attendant le merge et l'ouverture d'un compte kosmtik sur le hub Docker, vous pouvez utiliser celui ci : nobletjulien/kosmtik
`docker run -v */absolute_path/to/your/project*:/project -p 6789 nobletjulien/kosmtik`

### CODE REVIEW

**Comment by yohanboniface** [comment]
Excellent!
@julien-noblet j'ai créé l'orga "kosmtik" sur le hub et je t'ai ajouté comme owner, tu peux te charger d'y uploader l'image? :)

File: README.md [hunk 1]
```diff
@@ -154,3 +153,10 @@ exports.LocalConfig = function (localizer, project) {
 - [kosmtik-mbtiles-export](https://github.com/kosmtik/kosmtik-mbtiles-export): export your project in MBTiles
 
 Run `node index.js plugins --available` to get an up to date list.
+
+
+## Usage with Docker
+
+Run `docker run -v */absolute_path/to/your/project*:/project -p 6789 kosmtik/kosmtik:lastest`
```
Comment (yohanboniface) [inline README.md]: typo: latest ;)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM node:0.10-onbuild
+
+VOLUME ["/project"]
+EXPOSE 6789
+
+CMD ["serve","/project/project.yml"]
```
Comment (yohanboniface) [inline Dockerfile]: Does that mean that the project file *needs* to be named `project.yml`, or is it overridable?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FROM node:0.10-onbuild
```
Comment (yohanboniface) [inline Dockerfile]: No 0.12 version around?

**Comment by yohanboniface** [comment]
If my style uses a local PostGIS database, am I able to connect to it? or is it only for remote vector tiles based styles?

File: README.md (hunk 1 above)
Comment (julien-noblet) [inline README.md]: Ooops :+1:

File: Dockerfile (hunk 2 above)
Comment (julien-noblet) [inline Dockerfile]: Not realy you can ovverrid the command and arguments:
`docker run -v /home/julien/my-super-project/:/project -p 6789 kosmtik/kosmtik:latest serve /project/youpla.yml`

File: Dockerfile (hunk 3 above)
Comment (julien-noblet) [inline Dockerfile]: 0.12 is available too.

**Comment by julien-noblet** [comment]
I will try to connect it to a PostGIS in another Docker container ^^
I haven't test it right now.

File: Dockerfile (hunk 3 above)
Comment (yohanboniface) [inline Dockerfile]: So let's use 0.12 :)

File: Dockerfile (hunk 2 above)
Comment (yohanboniface) [inline Dockerfile]: Cool, maybe you can this in the Docker section of the README? :)

**Comment by yohanboniface** [comment]
> I will try to connect it to a PostGIS in another Docker container ^^

Not necessarily in another container: let's say the `project.yml` file defines a postgis layer (for example [this one](https://github.com/opendatasoft/transilien-network-carto/blob/master/project.yml#L36)), is there a way (maybe using a localconfig.js file) to connect to the host PostGIS DB?

cc @jpetazzo (summary: connect host PostGIS DB from a container)

File: Dockerfile (hunk 3 above)
Comment (julien-noblet) [inline Dockerfile]: done :+1:

**Comment by jpetazzo** [comment]
@yohanboniface: in your example, where does the PostGIS DB run? 

- in the same container: just connect to localhost
- in a different container on the same host: use a link
- on the host: connect to the host address (you can use `--add-host` to abstract that)
- on a different host: use `--add-host` if it will always be on the default port; otherwise, inject host+port with env vars and patch the file (e.g. with `sed`) before starting the service

**Comment by yohanboniface** [comment]
@jpetazzo Thanks dude :)
So there is no default hostname for the *host*? I mean, one needs to know the host IP event when using `--add-host`, as I understand, right? Is there a way to imagine that the docker container will always map `parenthost` (or whatever) to the host IP?

In Kosmtik, there is no clear use case if PostGIS DB is on another container or on the host itself; power users, docker experimented, will certainly use another container, while less advanced users may just want to connect to the host PostGIS DB that they may have already filled in (Kosmtik is a NodeJS application to create maps, so it targets designers more than developers, and every path that makes their life easier is good to go :) )

@julien-noblet maybe in the future we can think about a "osm2pgsql" docker container? So one would do something like

    docker run -p 6789 kosmtik/osm2pgsql -G path/to/data.osm.pbf --hstore --create

What do you think?
And maybe even better:

    docker run -p 6789 kosmtik/osm2pgsql france
    docker run -p 6789 kosmtik/osm2pgsql bretagne

and then mapping to Geofabrik downloads internally. Thoughts? :)

**Comment by jpetazzo** [comment]
@yohanboniface: unfortunately, there is no default hostname for the host itself. (It's been considered a few times but never implemented; I don't remember if there was a particular reason other than "nobody cared to submit a PR yet" :-))

I think I would connect to `$PGHOST:$PGPORT`, and have those variables default to `pgsql` and `5432`. This would allow flexible customization *and* a default behavior that would work with links out of the box. (Using a connection string is fine as well, if that's what kosmtik expects.)

**Comment by julien-noblet** [comment]
I'm looking for connect to openfirmware/postgres-osm

**Pull request closed without merging**